### PR TITLE
Fix broken encoding conversion after buffer resize

### PR DESCRIPTION
**Pull request by rianmcguire**
Reinitialize message descriptor (MQMD) if we're retrying after a buffer resize.

Some of the MQMD fields, like `Encoding` are both input AND output fields. If the request fails due to `MQRC_TRUNCATED_MSG_FAILED`, the conversion won't happen and `md.Encoding` will be set to the pre-conversion encoding. If we re-send the same `md` value, we end up requesting that encoding instead of the native encoding we wanted.

### CODE REVIEW

**Review by bazlo — APPROVED** [review]
Great detective work 🔍 

You deserve an award 🏅 of some sort!

File: ext/wmq_queue.c [hunk 1]
```diff
@@ -689,10 +689,9 @@ VALUE Queue_get(VALUE self, VALUE hash)
     MQLONG   flag;
     MQLONG   messlen;                /* message length received       */
 
-    MQMD     md = {MQMD_DEFAULT};    /* Message Descriptor            */
+    static const MQMD md_def = {MQMD_DEFAULT};
```
Comment (tomdalling) [inline ext/wmq_queue.c:692]: `md_def` is never mutated, right? If it was, this wouldn't be thread safe.

File: ext/wmq_queue.c [hunk 2]
```diff
@@ -772,28 +775,33 @@ VALUE Queue_get(VALUE self, VALUE hash)
               &pq->comp_code,      /* completion code                   */
               &pq->reason_code);   /* reason code                       */
 
-        /* report reason, if any     */
-        if (pq->reason_code != MQRC_NONE)
+        if (pq->reason_code == MQRC_TRUNCATED_MSG_FAILED)
```
Comment (tomdalling) [inline ext/wmq_queue.c:778]: Looks like this replaces the `pq->reason_code != MQRC_NONE` check. Is that intended?

**Review by tomdalling — COMMENTED** [review]
Looks good

File: ext/wmq_queue.c (hunk 2 above)
Comment (rianmcguire) [reply ext/wmq_queue.c:778]: Yes. The previous redundant logic was:
```
if (pq->reason_code != MQRC_NONE)
    if (pq->reason_code == MQRC_TRUNCATED_MSG_FAILED)
```

File: ext/wmq_queue.c (hunk 1 above)
Comment (rianmcguire) [reply ext/wmq_queue.c:692]: Correct. The only thing we do with it is copy it into `md`.

**Pull request merged**